services/shhext/chat/encryption_test.go

```diff
@@ -314,6 +314,213 @@ func (s *EncryptionServiceTestSuite) TestConversation() {
 	s.Equal(cleartext2, decryptedPayload1, "It correctly decrypts the payload using X3DH")
 }
 
+// Previous implementation allowed max maxSkip keys in the same receiving chain
+// leading to a problem whereby dropped messages would accumulate and eventually
+// we would not be able to decrypt any new message anymore.
+// Here we are testing that maxSkip only applies to *consecutive* messages, not
+// overall.
+func (s *EncryptionServiceTestSuite) TestMaxSkipKeys() {
+	bobText := []byte("text")
+
+	bobKey, err := crypto.GenerateKey()
```

Comment:
Thinking out loud, should we extract the shared test setup ("arrange") code into a helper method to keep the test code short and to the point? The logic seems to be repeated often enough to warrant it.

Reply:
Ok, if you don't mind I will do that in a separate PR, getting a bit busy here as I have 5 PRs pending :) , I also will add some tests for multi-device as it's not properly covered currently.